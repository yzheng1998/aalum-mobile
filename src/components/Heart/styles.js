import styled from 'styled-components'
import { themeGet } from 'styled-system'
import Icon from '../Icon'

export const Icn = styled(Icon)`
  color: ${props =>
    props.active
      ? themeGet('colors.gradientLeft')
      : themeGet('colors.lightGrey')};
  font-size: 32px;
`
