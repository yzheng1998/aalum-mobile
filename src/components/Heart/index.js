import React, { Component } from 'react'
import { Icn } from './styles'

class Heart extends Component {
  render() {
    const { active } = this.props
    return <Icn active={active} name={active ? 'heart' : 'heart-empty'} />
  }
}

export default Heart
