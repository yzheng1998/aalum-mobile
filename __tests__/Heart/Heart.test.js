import React from 'react'
import Heart from '../../src/components/Heart'
import mountWithTheme from '../helpers/theme'

describe('Heart', () => {
  test('renders correctly', () => {
    const wrapper = mountWithTheme(<Heart active />)
    expect(wrapper).toMatchSnapshot()

    wrapper.setProps({ active: false })
    expect(wrapper).toMatchSnapshot()
  })
})
