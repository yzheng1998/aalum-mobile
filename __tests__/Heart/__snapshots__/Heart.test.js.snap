// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heart renders correctly 1`] = `
<ThemeProvider
  theme={
    Object {
      "colors": Object {
        "black": "#4F4F4F",
        "border": "#BDBDBD",
        "buttonShadow1": "rgba(255, 255, 255, 0) ",
        "buttonShadow2": "rgba(249, 249, 249, 0.9)",
        "charcoal": "#333333",
        "darkGrey": "#8F8F8F",
        "darkShadow": "rgba(0,0,0,0.1)",
        "darkerGrey": "#828282",
        "error": "#DA2752",
        "gradientLeft": "#DA2752",
        "gradientRight": "#E935F8",
        "green": "#219653",
        "grey": "#A8A8A8",
        "inactiveText": "#ADADAD",
        "lightGrey": "#A0A0A0",
        "lightShadow": "rgba(0,0,0,0.16)",
        "offWhite": "#F9F9F9",
        "pureBlack": "#000000",
        "shadow": "rgba(0, 0, 0, 0.15)",
        "white": "#FFFFFF",
        "whiteGrey": "#E0E0E0",
      },
      "fonts": Object {
        "bodyText": "Avenir-Medium",
        "heading": "Montserrat-Bold",
        "title": "Avenir-Heavy",
      },
    }
  }
>
  <Heart
    active={true}
  >
    <Styled(_default)
      active={true}
      name="heart"
    >
      <StyledNativeComponent
        active={true}
        forwardedClass={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "displayName": "Styled(_default)",
            "inlineStyle": InlineStyle {
              "rules": Array [
                "
  color: ",
                [Function],
                ";
  font-size: 32px;
",
              ],
            },
            "render": [Function],
            "styledComponentId": "StyledNativeComponent",
            "target": [Function],
            "withComponent": [Function],
            Symbol(Symbol.hasInstance): [Function],
          }
        }
        forwardedRef={null}
        name="heart"
      >
        <_default
          active={true}
          name="heart"
          style={
            Array [
              Object {
                "color": "#DA2752",
                "fontSize": 32,
              },
            ]
          }
        >
          <Icon
            active={true}
            allowFontScaling={false}
            name="ios-heart"
            size={12}
            style={
              Array [
                Object {
                  "color": "#DA2752",
                  "fontSize": 32,
                },
              ]
            }
          >
            <Text
              active={true}
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 12,
                  },
                  Array [
                    Object {
                      "color": "#DA2752",
                      "fontSize": 32,
                    },
                  ],
                  Object {
                    "fontFamily": "Ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              <Text
                active={true}
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": undefined,
                      "fontSize": 12,
                    },
                    Array [
                      Object {
                        "color": "#DA2752",
                        "fontSize": 32,
                      },
                    ],
                    Object {
                      "fontFamily": "Ionicons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
            </Text>
          </Icon>
        </_default>
      </StyledNativeComponent>
    </Styled(_default)>
  </Heart>
</ThemeProvider>
`;

exports[`Heart renders correctly 2`] = `
<ThemeProvider
  active={false}
  theme={
    Object {
      "colors": Object {
        "black": "#4F4F4F",
        "border": "#BDBDBD",
        "buttonShadow1": "rgba(255, 255, 255, 0) ",
        "buttonShadow2": "rgba(249, 249, 249, 0.9)",
        "charcoal": "#333333",
        "darkGrey": "#8F8F8F",
        "darkShadow": "rgba(0,0,0,0.1)",
        "darkerGrey": "#828282",
        "error": "#DA2752",
        "gradientLeft": "#DA2752",
        "gradientRight": "#E935F8",
        "green": "#219653",
        "grey": "#A8A8A8",
        "inactiveText": "#ADADAD",
        "lightGrey": "#A0A0A0",
        "lightShadow": "rgba(0,0,0,0.16)",
        "offWhite": "#F9F9F9",
        "pureBlack": "#000000",
        "shadow": "rgba(0, 0, 0, 0.15)",
        "white": "#FFFFFF",
        "whiteGrey": "#E0E0E0",
      },
      "fonts": Object {
        "bodyText": "Avenir-Medium",
        "heading": "Montserrat-Bold",
        "title": "Avenir-Heavy",
      },
    }
  }
>
  <Heart
    active={true}
  >
    <Styled(_default)
      active={true}
      name="heart"
    >
      <StyledNativeComponent
        active={true}
        forwardedClass={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "displayName": "Styled(_default)",
            "inlineStyle": InlineStyle {
              "rules": Array [
                "
  color: ",
                [Function],
                ";
  font-size: 32px;
",
              ],
            },
            "render": [Function],
            "styledComponentId": "StyledNativeComponent",
            "target": [Function],
            "withComponent": [Function],
            Symbol(Symbol.hasInstance): [Function],
          }
        }
        forwardedRef={null}
        name="heart"
      >
        <_default
          active={true}
          name="heart"
          style={
            Array [
              Object {
                "color": "#DA2752",
                "fontSize": 32,
              },
            ]
          }
        >
          <Icon
            active={true}
            allowFontScaling={false}
            name="ios-heart"
            size={12}
            style={
              Array [
                Object {
                  "color": "#DA2752",
                  "fontSize": 32,
                },
              ]
            }
          >
            <Text
              active={true}
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 12,
                  },
                  Array [
                    Object {
                      "color": "#DA2752",
                      "fontSize": 32,
                    },
                  ],
                  Object {
                    "fontFamily": "Ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              <Text
                active={true}
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": undefined,
                      "fontSize": 12,
                    },
                    Array [
                      Object {
                        "color": "#DA2752",
                        "fontSize": 32,
                      },
                    ],
                    Object {
                      "fontFamily": "Ionicons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
            </Text>
          </Icon>
        </_default>
      </StyledNativeComponent>
    </Styled(_default)>
  </Heart>
</ThemeProvider>
`;
